### PR TITLE
[Tests] Add missing Python import checks

### LINKED ISSUE #526: Check we have cases in `test_imports.py` for recently added modules
## What

Ensure `test_imports.py` covers all public python.

## Why

We forgot to add these tests for some recent additions.

### PR DESCRIPTION
**Pull request by foundrytom**
Accidentally omitted during recent additions/changes.

Closes #526

### CODE REVIEW

File: tests/python/openassetio/test_imports.py [hunk 1]
```diff
@@ -115,3 +121,18 @@ def test_importing_PluginSystemManagerInterfaceFactory_succeeds(self):
 
     def test_importing_PluginSystemPlugin_succeeds(self):
         from openassetio.pluginSystem import PluginSystemPlugin
+
+
+class Test_test_imports:
```
Comment (feltech) [inline tests/python/openassetio/test_imports.py:126]: [question] Looks all good, I'm just wondering what the rule is for what should be included in this test module.  

At first I thought the intention was to test classes have been hoisted successfully in `__init__.py`s. But the tests here and below break that rule.

**Review by feltech — APPROVED** [review]

File: tests/python/openassetio/test_imports.py (hunk 1 above)
Comment (foundrytom) [reply tests/python/openassetio/test_imports.py:126]: Originally, the main intent of these tests was to catch cyclic imports etc... so I guess it should be that any individually importable module is importable, but it also doubles up testing C++ hoists now too. So I guess its:

- Any individual python defined module
- Any C++ hoisted module.

Comment (foundrytom) [reply tests/python/openassetio/test_imports.py:126]: Can you think of anything else? Ill add a note to such an effect at the top

Comment (feltech) [reply tests/python/openassetio/test_imports.py:126]: In which case we're missing `test.manager.harness`/`apiComplianceSuite`?  Or is it that these are somewhat "private"?

Comment (foundrytom) [reply tests/python/openassetio/test_imports.py:130]: Added notes on coverage [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/534/commits/52d92df5eed34eb32374273fc9e21d5247213b8f), hows that? Added missing tests [https://github.com/OpenAssetIO/OpenAssetIO/pull/534/commits/be3a4e649a7c969296ec27b1515a8187e49e87d0] - somehow I'd totally spaced them.

Comment (feltech) [reply tests/python/openassetio/test_imports.py:130]: LGTM, thanks.

**Pull request merged**